### ISSUE TITLE
<include> behaves differently based on template path source

### ISSUE THREAD

**Issue by jackhorton** [raptorjs/marko#96]
When `<include>` is given a hard-coded path, it can be relative, as said in the docs, or absolute. However, when it is given a variable path, as in `<include template="$data.templatePath" />`, it must be absolute (it will error out if relative).

**Works**
```html
<include template="../relative/path/to/template.marko" />
```

**Works**
```html
<var name="templatePath" value="'/absolute/path/to/template.marko'" />
<include template="$templatePath" />
```

**Works**
```html
<include template="/absolute/path/to/template.marko" />
```

**Does Not Work**
```html
<var name="templatePath" value="'../relative/path/to/template.marko'" />
<include template="$templatePath" />
```

It seems as if the issue is that hard-coded paths are `require.resolve`'d, while variable paths are not. Also, the docs could use an update, since it says that the template path must be relative, even for hard-coded paths.

**Comment by patrick-steele-idem** [comment]
Hi @jackhorton, this is not clarified in the docs, but if the `template` attribute is  dynamic expression (not a static string) then the expression should resolve to an already _loaded_ template. You can load the template in the template itself as shown below:

```xml
<require module="../relative/path/to/template.marko" var="template"/>
<include template="$template" />
```

However, if it usually cleaner for the include template to be loaded by the custom tag/controller and for it to be provided as part of the view model to the template.

```xml
<include template="$data.myLoadedIncludeTemplate" />
```

I'll update the docs, but let me know if that doesn't work for you.

**Comment by patrick-steele-idem** [comment]
Hi @jackhorton, just an update: I put in a minor tweak to allow the `template` attribute to be a dynamic path that can be relative. It is still recommended to preload the include target and pass a reference to the fully loaded template. This mainly because JavaScript module bundlers that do static code analysis will not find dynamically required templates.

New version published: `marko@2.7.4`

Please let me know if you have any questions or still have issues. Thanks.
